modules/checklist/steps/step-base.php

```diff
@@ -191,4 +196,36 @@ public function get_step_initial_progress() : array {
 	private function set_step_progress() : void {
 		$this->module->set_step_progress( $this->get_id(), $this->user_progress );
 	}
+
+	public function page_exists( $params ) : bool {
+		$query = new \WP_Query( $params );
+		return $query->found_posts;
+	}
+
+	public function get_elementor_create_new_page_url(): string {
+		return $this->wordpress_adapter->add_query_arg( [
+			'active-document' => self::SITE_SETTINGS_TAB,
+			'active-tab' => self::SITE_IDENTITY_TAB,
+		], Plugin::$instance->documents->get_create_new_post_url( 'page' ) );
+	}
+
+	public function get_site_settings_url_config(): string {
+		$existing_elementor_page = $this->get_elementor_page();
+		$site_settings_url = ! empty( $existing_elementor_page )
+			? $this->get_elementor_edit_url( $existing_elementor_page->ID )
+			: $this->get_elementor_create_new_page_url();
+
+		return $site_settings_url;
+	}
+
+	public function get_elementor_edit_url( int $post_id ): string {
+		$active_kit_id = Plugin::$instance->kits_manager->get_active_id();
+		$document = Plugin::$instance->documents->get( $post_id );
+
+		if ( ! $document ) {
+			return '';
+		}
+
+		return add_query_arg( [ 'active-document' => $active_kit_id ], $document->get_edit_url() );
+	}
```

Comment:
Will these be re-used in other steps as well? Just trying to understand if it belongs in the base. It is currently specific to your step. If we need it in another step we can put it in a base.